UserAuthAPI/Controllers/AuthController.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using UserAuthAPI.Requests;
+
+namespace UserAuthAPI.Controllers
+{
+    [Route("api/auth")]
+    [ApiController]
+    public class AuthController : ControllerBase
+    {
+        private readonly ApplicationDBContext _dbContext;
+        private readonly JWTUtil _util;
+        public AuthController(ApplicationDBContext dBContext, JWTUtil util)
+        {
+            _dbContext = dBContext;
+            _util = util;
+        }
+
+        [HttpPost("register")]
+        public async Task<ActionResult> register(RegisterRequest registerRequest)
+        {
+            var role = await _dbContext.UserRoles.FindAsync(registerRequest.UserRoleId);
+            if (role is null) {
+                return BadRequest();
+            }
+
+            Model.User user = new Model.User
```

Comment:
Need to inclue validations for email exists, username exists and model validations for email format and field sizes. Best if the database handling logic is moved outside of the controller to a separate layer.